### PR TITLE
Dashboard rewrite june 2022

### PR DESCRIPTION
**Pull request by prosif**
~~wip~~
Half step towards a rewrite of the dashboard. Still a bunch of bugs and busted logic in the version handling and downloading games, but generally structured closer to what it should be. 

PR was getting huge so I figure we can get this merged while I fix the remaining bugs and get the local games working over the next few weeks. 

I plan to test game publishing as well as our build service to publish new builds of homegames starting with this PR once we get it merged.

So basically ignore bugs. If you see confusing logic, feel free to point it out but don't spend too much time banging your head. I'll likely rewrite confusing pieces as I fix remaining bugs.

### CODE REVIEW

**Comment by prosif** [comment]
@YazeedLoonat 

More info in the description, but basically this PR got huge so I put it up now to be merged so I can test other stuff including builds of the HG executable starting with this PR.

I'll fix bugs and rewrite stuff in smaller changes (and corresponding builds) over the next month as kind of a pre-pre-release time where we test publishing and search and just testing overall

**Comment by prosif** [comment]
oh yeah also ignore failing pipeline, I image I'll need to rewrite tests in the future

**Comment by prosif** [comment]
@YazeedLoonat pushed a bunch more changes to clean up and fix stuff. Dashboard is working better overall now especially with game downloads. We can go over these in person or however over the next few days but at this point I think this is mergeable.

Some games still don't work correctly and there are still some rough edges here and there, but I'll be following up with various smaller fixes throughout the month.

**Comment by prosif** [comment]
also still ignore failing tests - I will follow up in a future PR to fix those

**Comment by prosif** [comment]
"ayy lmao" - George Washington
